File: 08-1-2-State-Solution/States/PartnerStateBase.cs
namespace State.States
{
    using System;

    internal abstract class PartnerStateBase : IPartnerState
    {
        protected PartnerStateBase(Partner partner)
        {
            Partner = partner;
        }

        public abstract void Talk();

        public abstract void Kiss();

        public abstract void Insult();

        protected Partner Partner { get; }

        protected void Print(string message) => Console.WriteLine(message);

        protected void SetState(IPartnerState newState) => Partner.State = newState;
    }
}

File: 10-DesignPatterns-Java/Solution/State/src/States/PartnerStateBase.java
package States;

public abstract class PartnerStateBase implements IPartnerState {
    protected Partner.Partner _partner;

    protected PartnerStateBase(Partner.Partner partner) {
        _partner = partner;
    }

    public abstract void talk();

    public abstract void kiss();

    public abstract void insult();

    protected void print(String message) {
        System.out.println(message);
    }

    protected void setState(IPartnerState newState) {
        _partner.setState(newState);
    }
}
